Userland/Libraries/LibC/complex.cpp

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2022, Peter Elliott <pelliott@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <complex.h>
+
+extern "C" {
+
+// https://pubs.opengroup.org/onlinepubs/9699919799/functions/creal.html
+float(crealf)(float complex z)
```

Comment:
This style of function declaration looks out of place

Reply:
This prevents `crealf` from being expanded as a macro. I think `clang-format` got rid of the space after `float`.
See this comment for why this was done like this:

>These are supposed to be the same name. Basically we prefer that these are always inlined as macros, because there is no reason to make a function call for what is likely accessing a value in a register.
>
>Implementing these as macros is allowed by c99:
>
>> 7.1.4p1: Any function declared in a header may be additionally implemented as a function-like macro defined in the header.
>
>The reason they are also declared as functions is because they need to have external linkage:
>
>>7.1.2p6: Any declaration of a library function shall have external linkage.

Reply:
A slightly less `:acid2:` approach would be to `#undef` these macros before the functions + add a comment explaining why that's needed.

Reply:
If you undef them, how would you call the macros? I have added comments though.

Reply:
Guess who didn't look hard enough 🤦‍♂️